text/0047-introduce-profile-context.md

```diff
@@ -0,0 +1,110 @@
+- Start Date: 2022-12-20
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/47
+- RFC Status: draft
+
+# Summary
+
+We want to introduce the concept of a profiling context, much like a tracing context, starting with
+transactions so we can easily relate a group of transactions back to the respective profile.
```

Comment:
On the Cocoa SDK, @armcknight opened a draft PR to change the relationship between transactions and profiles because of problems with DS. Currently, a profile can relate to a group of transactions. The [draft PR](https://github.com/getsentry/sentry-cocoa/pull/2545) proposes to change this so that a profile always relates to one transaction only. If there are multiple transactions in parallel, each one gets its own profiling data, which will be duplicated for the time they overlap. @Zylphrex, I think it's best to sync with @armcknight before continuing with this RFC.

Reply:
Yes, the change @armcknight is making is expected and will be the way all SDKs handle parallel transactions (by duplicating the profile data).

The end result of this is expected to be 
- each transaction will have the corresponding profiling data from the same time window
- the transaction and the profile should be sent in the same envelope (to deal with the DS problem)
- if there are parallel transactions, we will duplicate the profiling data for each transaction (by trimming the profile for the transaction's time window)

The *new* change from this RFC is that the transaction will contain a profile context that contains the information for the corresponding profile so that it can be used to relate a transaction back to the profile easily.

Reply:
That is correct. There will be a 1:1 relationship between transactions and profiles. And this context will power the relation from transactions to profiles.